vmm/src/device_manager.rs

```diff
@@ -3027,14 +3043,33 @@ impl DeviceManager {
                 .map_err(DeviceManagerError::UpdateMemoryForVirtioDevice)?;
         }
 
+        self.update_memory(
+            &Range {
+                iova: new_region.start_addr().raw_value(),
+                size: new_region.len() as u64,
+                user_addr: new_region.as_ptr() as u64,
+            },
+            true,
+        )
+    }
+
+    pub fn update_memory(&self, range: &Range, add: bool) -> DeviceManagerResult<()> {
```

Comment:
I would prefer add to be an enum of AddMemoryRange/RemoveMemoryRange

Reply:
Of course, it's definitely more rusty!